From f2ed12faa4ecb5468f6396f1695a3ca5e2e13b84 Mon Sep 17 00:00:00 2001
From: Stef Walter <stefw@redhat.com>
Date: Fri, 24 Jul 2015 17:25:32 +0200
Subject: [PATCH] service: Increase timeout when calling PackageKit

Use a timeout of a day ... PackageKit should return from a call
sooner than that. If it crashes we'll get a response right away.
---
 service/realm-packages.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/service/realm-packages.c b/service/realm-packages.c
index c32da44..9da852c 100644
--- a/service/realm-packages.c
+++ b/service/realm-packages.c
@@ -24,6 +24,8 @@
 
 #include <glib/gi18n.h>
 
+#define CALL_TIMEOUT (24 * 60 * 60 * 1000)
+
 static gboolean
 packages_check_paths (const gchar **paths,
                             GDBusMethodInvocation *invocation)
@@ -226,7 +228,7 @@ on_set_hints (GObject *source,
 		                        transaction->parameters,
 		                        G_VARIANT_TYPE ("()"),
 		                        G_DBUS_CALL_FLAGS_NO_AUTO_START,
-		                        -1, g_task_get_cancellable (task),
+		                        CALL_TIMEOUT, g_task_get_cancellable (task),
 		                        on_method_done, g_object_ref (task));
 	}
 
@@ -276,7 +278,7 @@ on_create_transaction (GObject *source,
 		                        g_variant_new ("(^as)", hints),
 		                        G_VARIANT_TYPE ("()"),
 		                        G_DBUS_CALL_FLAGS_NO_AUTO_START,
-		                        -1, g_task_get_cancellable (task),
+		                        CALL_TIMEOUT, g_task_get_cancellable (task),
 		                        on_set_hints, g_object_ref (task));
 	}
 
@@ -310,7 +312,7 @@ package_transaction_create (const gchar *method,
 	                        g_variant_new ("()"),
 	                        G_VARIANT_TYPE ("(o)"),
 	                        G_DBUS_CALL_FLAGS_NONE,
-	                        -1, cancellable,
+	                        CALL_TIMEOUT, cancellable,
 	                        on_create_transaction, g_object_ref (task));
 }
 
-- 
2.4.3

